### PR TITLE
FIX-#4966: Fix `to_timedelta` to return Series instead of TimedeltaIndex

### PR DESCRIPTION
**Pull request by Billy2551**
Signed-off-by: Bill Wang <billiam@ponder.io>

## What do these changes do?

Added implementation for `to_timedelta` instead of using pandas version. Previously returned incorrect output when using a Modin Series as input.

Added tests for `to_timedelta` with Modin Series.

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #? 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

File: modin/core/storage_formats/pandas/query_compiler.py
Comment (mvashishtha) [inline modin/core/storage_formats/pandas/query_compiler.py:1384]: I think we can pass `dtypes="timedelta64[ns]"` because the pandas docs say that the output is "Series of timedelta64 dtype". I can't find any `timedelta64` type besides `timedelta64[ns]`, so I guess that's the type to pass. Setting dtypes ahead of time means that we don't have to figure them out from the partitions later.

File: modin/pandas/test/test_general.py
Comment (mvashishtha) [inline modin/pandas/test/test_general.py:810]: We should test float and int data, at least, with enough data to fill a few partitions. `test_data` in the utils has exactly those types. We can adapt the code here to parametrize by the data type-- maybe we don't need a method as complicated as `create_test_series`. https://github.com/modin-project/modin/blob/0a2c0de4451f7e2e8f337a9478d7595473aa348e/modin/pandas/test/test_series.py#L210-L212

**Review by pyrito — COMMENTED** [review]
@Billy2551 looks like there are some documentation errors. Could you take a look at them?

**Comment by mvashishtha** [comment]
> looks like there are some documentation errors. Could you take a look at them?

@Billy2551 I recommend running locally the docstyle command from the [failed CI job](https://github.com/modin-project/modin/actions/runs/3115012026/jobs/5051480045): 

```shell
  python scripts/doc_checker.py --add-ignore=D101,D102,D103,D105 --disable-numpydoc \
    modin/pandas/dataframe.py modin/pandas/series.py \
    modin/pandas/groupby.py \
    modin/pandas/series_utils.py modin/pandas/general.py \
    modin/pandas/plotting.py modin/pandas/utils.py \
    modin/pandas/iterator.py modin/pandas/indexing.py \
```

File: modin/core/storage_formats/base/query_compiler.py
Comment (vnlitvinov) [inline modin/core/storage_formats/base/query_compiler.py:1333]: I don't see those arguments in the function signature, it should error out in the docs

File: modin/core/storage_formats/pandas/query_compiler.py
Comment (vnlitvinov) [inline modin/core/storage_formats/pandas/query_compiler.py:1386]:
```suggestion
            s.squeeze(axis=1), *args, **kwargs
```
is that right? I would prefer using named arguments for better readability

File: modin/pandas/general.py
Comment (vnlitvinov) [inline modin/pandas/general.py:773]: this feels really weird - is there a reason why we should return a pandas object from this method?.. I would prefer to return Modin Series here at all cases

Comment (Billy2551) [reply modin/pandas/general.py]: Fixed to convert pandas Series results to modin Series

File: modin/core/storage_formats/pandas/query_compiler.py
Comment (Billy2551) [reply modin/core/storage_formats/pandas/query_compiler.py]: Fixed.

File: modin/core/storage_formats/base/query_compiler.py
Comment (Billy2551) [reply modin/core/storage_formats/base/query_compiler.py]: Fixed.

File: docs/release_notes/release_notes-0.16.0.rst
Comment (pyrito) [inline docs/release_notes/release_notes-0.16.0.rst:56]:
```suggestion
  * FIX-#4966: Fix `to_timedelta` to return Series instead of TimedeltaIndex (#5028 )
```

File: modin/core/storage_formats/base/query_compiler.py
Comment (mvashishtha) [inline modin/core/storage_formats/base/query_compiler.py:1337]:
```suggestion
    def to_timedelta(self, unit="ns", errors="raise"):
        """
        Convert argument to timedelta.

        Parameters
        ----------
        unit : str, default: "ns"
            Denotes the unit of the arg for numeric arg. Defaults to "ns".
        errors : {"ignore", "raise", "coerce"}, default: "raise"
```

I don't think pandas supports `*args` or `**kwargs`

File: modin/core/storage_formats/base/query_compiler.py
Comment (mvashishtha) [inline modin/core/storage_formats/base/query_compiler.py:1346]:
```suggestion
        return SeriesDefault.register(pandas.to_timedelta)(self, unit=unit, errors=errors)
```

File: modin/pandas/general.py
Comment (mvashishtha) [inline modin/pandas/general.py:776]:
```suggestion
    return pandas.to_timedelta(arg, unit=unit, errors=errors)
```

```suggestion
```

According to the documentation, return type is series if and only if the input is a series. So I don't think we could hit this case.

File: modin/pandas/test/test_general.py
Comment (mvashishtha) [inline modin/pandas/test/test_general.py:807]: Could you add another test case where we pass:

- a list-like of ints, like `[1]`
- a list-like of invalid strings, like `['a']`. This should raise `ValueError`.
- a valid scalar, like `1`
- an invalid scalar, like '"a"'

We can do something like `@pytest.mark.parametrize('arg', [[1, 2], ["a"], 1, "a"], ids=["list_of_ints", "list_of_invalid_strings", "scalar", "invalid_scalar"])`

**Review by mvashishtha — DISMISSED** [review]
LGTM @Billy2551 ! Thank you.

**Review by pyrito — APPROVED** [review]

File: modin/pandas/general.py
Comment (vnlitvinov) [inline modin/pandas/general.py:773]: This is again returning some pandas object, not Modin, is that really what we want?

I think our stance on API is we mimic pandas, but we use our own types everywhere.

File: modin/pandas/test/test_general.py
Comment (vnlitvinov) [inline modin/pandas/test/test_general.py:832]: let's use `eval_general()` here, too - it takes care of comparing exceptions if they raise (plus it correctly waits for async Modin calls as they only raise exceptions when we ask for the result):
```suggestion
    def make_frame(lib):
        series = lib.Series(
            next(iter(data.values())) if isinstance(data, dict) else data
        )
        return lib.to_timedelta(series).to_frame(name="timedelta")

    eval_general(pd, pandas, make_frame)

```

**Review by vnlitvinov — CHANGES_REQUESTED** [review]

**Review by mvashishtha — DISMISSED** [review]
@Billy2551 I have a minor comment, but LGTM once @vnlitvinov 's comments are resolved.

**Review by mvashishtha — APPROVED** [review]
Thanks @Billy2551

**Review by pyrito — APPROVED** [review]

**Review by vnlitvinov — APPROVED** [review]
LGTM, thanks @Billy2551!

**Review by YarShev — APPROVED** [review]

**Pull request merged**
